src/TRestAxionXrayWindow.cxx

```diff
@@ -0,0 +1,318 @@
+
+    fclose(fin);
+}
+
+///////////////////////////////////////////////
+/// \brief It returns the window transmission probability for the given energy (in keV)
+/// and window position, using energy linear interpolation.
+///
+/// For the case of patterned window (stripped or grid), it will return 1 if the strip is
+/// not hitted.
+///
+Double_t TRestAxionXrayWindow::GetTransmission(Double_t energy, Double_t x, Double_t y) {
+    if (fEnergy.size() == 0) ReadMaterial();
+
+    if ((x - fCenter.X()) * (x - fCenter.X()) + (y - fCenter.Y()) * (y - fCenter.Y()) > fRadius * fRadius)
+        return 0;
+
+    if (!HitsPattern(x, y)) return 1.;
+
+    Double_t energyIndex = GetEnergyIndex(energy);
+
+    // Transmission
+    double y2 = fTransmission[energyIndex + 1];
+    double y1 = fTransmission[energyIndex];
+
+    // Normalized field
+    double x2 = fEnergy[energyIndex + 1];
+    double x1 = fEnergy[energyIndex];
+
+    double m = (y2 - y1) / (x2 - x1);
+    double n = y1 - m * x1;
+
+    if (m * energy + n < 0) {
+        ferr << "TRestAxionXrayWindow::GetAbsorptionCoefficient. Negative coefficient!" << endl;
+        cout << "y2 : " << y2 << " y1 : " << y1 << endl;
+        cout << "x2 : " << x2 << " x1 : " << x1 << endl;
+        cout << "m : " << m << " n : " << n << endl;
+        cout << "E : " << energy << " bin : " << energyIndex << endl;
+        GetChar();
+    }
+
+    return (m * energy + n);
+}
+
+///////////////////////////////////////////////
+/// \brief It returns true if the window pattern is hitted. False otherwise.
+///
+Bool_t TRestAxionXrayWindow::HitsPattern(Double_t x, Double_t y) {
+    if (fWindowType == "stripped") {
+        Double_t xEval = fPatternWidth / 2. + x - fPatternOffset;
+
+        if (xEval > 0) {
+            while (xEval > fPatternGap) xEval -= fPatternGap;
+        } else {
+            while (xEval < 0) xEval += fPatternGap;
+        }
+
+        if (xEval > fPatternWidth) {
+            return false;
+        }
+    } else if (fWindowType == "grid") {
+        Double_t xEval = fPatternWidth / 2. + x - fPatternOffset;
+
+        if (xEval > 0) {
+            while (xEval > fPatternGap) xEval -= fPatternGap;
+        } else {
+            while (xEval < 0) xEval += fPatternGap;
+        }
+
+        if (xEval < fPatternWidth) return true;
+
+        Double_t yEval = fPatternWidth / 2. + y - fPatternOffset;
+
+        if (yEval > 0) {
+            while (yEval > fPatternGap) yEval -= fPatternGap;
+        } else {
+            while (yEval < 0) yEval += fPatternGap;
+        }
+
+        if (yEval < fPatternWidth) return true;
+        return false;
+    }
+
+    return true;
+}
+
+///////////////////////////////////////////////
+/// \brief It returns the vector element index, from `fEnergy`, that is just below the given input energy.
+///
+Int_t TRestAxionXrayWindow::GetEnergyIndex(Double_t energy) {
+    for (unsigned int n = 0; n < fEnergy.size(); n++)
```

Comment:
Avoid `unsigned int` if you need to do subtractions with them. You don't want an underflow to happen. Also means you get a "nice" implicit type cast from `n` ⇒ whatever the compiler deems `n - 1` to be ⇒ `Int_t`.

Reply:
Fixed at f488660b4